pkg/network/node/ovscontroller.go

```diff
@@ -267,6 +268,17 @@ func (oc *ovsController) SetupOVS(clusterNetworkCIDR []string, serviceNetworkCID
 
 	// Table 10: VXLAN ingress filtering; filled in by AddHostSubnetRules()
 	// eg, "table=10, priority=100, tun_src=${remote_node_ip}, actions=goto_table:30"
+	// Swap overlay dst MAC for ingress OVN-K hybrid overlay traffic
+	if oc.localMAC == "" {
+		return fmt.Errorf("failed to find the MAC address for local IP %s", oc.localIP)
+	}
+	if err := oc.ensureTunMAC(); err != nil {
+		return err
+	}
+	// set the dst MAC for traffic from OVN-K nodes to local node
+	otx.AddFlow("table=10, priority=210, ip, nw_dst=%s, dl_dst=%s, actions=set_field:%s->eth_dst, resubmit:10", localSubnetGateway, oc.localMAC, oc.tunMAC)
```

Comment:
I believe if a host-level process or hostnetwork pod on one node sends to a pod IP on another node, that traffic will end up with a source IP of the `localSubnetGateway` on the source node. Adding this rule would break that traffic.

If there isn't anything else you can use to recognize this traffic, then you might need to use conntrack, and set `ct_mark` or `ct_label` in the following rule, so that this rule can then recognize the packet as being a reply to a packet rewritten by that rule...

Alternatively there could be different rules for sdn nodes and ovn-k nodes.

Also, why do you need the `resubmit`? If the answer is "because the rewriting doesn't take effect otherwise", then please put that in the comment.

Reply:
This rule matches both dst IP and MAC. So, only the packets with dst MAC equal to the node interface MAC will hit this rule. Such ingress traffic can only come from OVN-K nodes. The traffic you mention will use the MAC of the `tun0` interface so that it won't hit this rule.